src/ncp/ncp_base.cpp

```diff
@@ -1370,7 +1370,7 @@ otError NcpBase::HandlePropertySet_SPINEL_PROP_NEST_STREAM_MFG(uint8_t aHeader)
     }
 #endif
 
-    otDiagProcessCmdLine(mInstance, string, output, sizeof(output));
+    SuccessOrExit(error = otDiagProcessCmdLine(mInstance, string, output, sizeof(output)));
```

Comment:
I am not sure about returning `error` from the `otDiagProcessCmdLine ` and doing `SuccessOrExit()` here.

If the processing of command fails (say, one of args is incorrect), the `output` is generated to indicate this error and here we still want to emit the spinel `SPINEL_CMD_PROP_VALUE_IS ` response back to host. 

If we do `SuccessOrExit()` then we never send spinel response and I think host will time out waiting for it.

Reply:
If the function `otDiagProcessCmdLine` returns error, the RCP Spinel will [return the error code](https://github.com/openthread/openthread/blob/main/src/ncp/ncp_base.cpp#L1049) to the host via `SPINEL_CMD_PROP_VALUE_IS(SPINEL_PROP_LAST_STATUS)`. The host will [parse the error code](https://github.com/openthread/openthread/blob/main/src/lib/spinel/radio_spinel_impl.hpp#L786) and return the error code to the function [PlatDiagProcess](https://github.com/openthread/openthread/blob/main/src/lib/spinel/radio_spinel_impl.hpp#L2083). There won't be a Spinel timeout issue.

Reply:
You are correct. This should work fine. Thanks.
(I guess it is been a while since I last looked at NCP/spinel related code 😄 I have forgotten so much about it)